saas/models.py

```diff
@@ -1320,6 +1328,15 @@ def role_on_subscriber(self, user, plan, role_descr=None):
     def valid_for(self, **kwargs):
         return self.filter(grant_key=None, request_key=None, **kwargs)
 
+    # This returns valid roles.
+    # Uncomment if this is the approach to take.
+    # Doesn't apply to existing queries since it isn't using the get_queryset method
+    def valid_roles(self):
+        """
+        Returns roles that are currently valid
+        """
+        return self.filter(Q(ends_at__isnull=True) | Q(ends_at__gt=datetime_or_now()))
```

Comment:
The filter parameters are correct but a method already exist to check for valid roles. See a few lines above `def valid_for`. The `Q(ends_at__isnull=True) | Q(ends_at__gt=datetime_or_now()` filter should be integrated into the `filter(grant_key=None, request_key=None, **kwargs)` statement above in `valid_for`.